src/net.cpp

```diff
@@ -665,9 +665,11 @@ int CNetMessage::readHeader(const char *pch, unsigned int nBytes)
         return -1;
     }
 
-    // reject messages larger than MAX_SIZE
-    if (hdr.nMessageSize > MAX_SIZE)
+    // reject if message has an invalid header
+    if (!hdr.IsValid(Params().MessageStart())) {
+        LogPrint(BCLog::NET, "INVALID HEADER DETECTED\n");
```

Comment:
There is already logging in

```
LogPrintf("CMessageHeader::IsValid(): (%s, %u bytes) nMessageSize > MAX_SIZE
```

I have no opinion on where to do it, but it should suffice to log only one line instead of two for a single error.

Reply:
Without the log entry in L670, we would only log the invalid message size (**not** invalid magic check and invalid command strings).
We could drop the invalid message size but I think this is one we should keep since it smells after DoS/dishonest peers.

After this PR, you will get two log entries in case of an message size larger then `MAX_SIZE`.
1. `LogPrintf("CMessageHeader::IsValid(): (%s, %u bytes) nMessageSize > MAX_SIZE\n", GetCommand(), nMessageSize);`
2. `LogPrint(BCLog::NET, "INVALID HEADER DETECTED\n");`

Which I think is acceptable.

Of course we could remove the first entry about the message size.